chain/processor.go

```diff
@@ -138,18 +95,12 @@ func (b *StatelessBlock) Execute(
 
 			// Commit results to parent [TState]
 			tsv.Commit()
-
-			// Update key cache
-			if len(toCache) > 0 {
-				cacheLock.Lock()
-				for k := range toCache {
-					cache[k] = toCache[k]
-				}
-				cacheLock.Unlock()
-			}
 			return nil
 		})
 	}
+	if err := f.HandleErrors(); err != nil {
```

Comment:
I think you'd want to call this after `e.Wait()` because we aren't guaranteed the fetcher is done before then.